providers/implementations/keymgmt/ml_dsa_kmgmt.c

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_dispatch.h>
+#include <openssl/core_names.h>
+#include <openssl/param_build.h>
+#include <openssl/self_test.h>
+#include "crypto/ml_dsa.h"
+#include "internal/param_build_set.h"
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+#include "prov/provider_ctx.h"
+
+static OSSL_FUNC_keymgmt_free_fn ml_dsa_free_key;
+static OSSL_FUNC_keymgmt_has_fn ml_dsa_has;
+static OSSL_FUNC_keymgmt_match_fn ml_dsa_match;
+static OSSL_FUNC_keymgmt_import_fn ml_dsa_import;
+static OSSL_FUNC_keymgmt_export_fn ml_dsa_export;
+static OSSL_FUNC_keymgmt_import_types_fn ml_dsa_imexport_types;
+static OSSL_FUNC_keymgmt_export_types_fn ml_dsa_imexport_types;
+static OSSL_FUNC_keymgmt_load_fn ml_dsa_load;
+static OSSL_FUNC_keymgmt_get_params_fn ml_dsa_get_params;
+static OSSL_FUNC_keymgmt_gettable_params_fn ml_dsa_gettable_params;
+static OSSL_FUNC_keymgmt_validate_fn ml_dsa_validate;
+static OSSL_FUNC_keymgmt_gen_init_fn ml_dsa_gen_init;
+static OSSL_FUNC_keymgmt_gen_cleanup_fn ml_dsa_gen_cleanup;
+static OSSL_FUNC_keymgmt_gen_set_params_fn ml_dsa_gen_set_params;
+static OSSL_FUNC_keymgmt_gen_settable_params_fn ml_dsa_gen_settable_params;
+
+#define ML_DSA_POSSIBLE_SELECTIONS (OSSL_KEYMGMT_SELECT_KEYPAIR)
```

Comment:
I find this "alias" counter productive - if it is just *one* value then we should be using just that one value and not making up another name for the same single value IMHO.